integration-tests/bats/sql-branch.bats

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+
+    dolt sql <<SQL
+CREATE TABLE test (
+    pk int primary key
+);
+
+INSERT INTO test VALUES (0),(1),(2);
+SQL
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
+@test "sql-branch: DOLT_BRANCH works" {
+    run dolt branch
+    [[ ! "$output" =~ "new_branch" ]] || false
+
+    run dolt sql -q "SELECT DOLT_BRANCH('new-branch')"
+    [ $status -eq 0 ]
+
+    # dolt sql -q "select dolt_branch() should not change the branch
```

Comment:
I think you can remove this comment because the behavior matches dolt branch on the CLI.